frb_codegen/src/library/integration/integrator.rs

```diff
@@ -1,24 +1,37 @@
-use crate::integration::utils::extract_dir_and_modify;
+use crate::integration::utils::{
+    compute_effective_path, extract_dir_and_modify, replace_file_content,
+};
 use crate::library::commands::flutter::{flutter_pub_add, flutter_pub_get};
 use crate::library::commands::format_dart::format_dart;
+use crate::misc::ProjectType;
 use crate::utils::dart_repository::get_dart_package_name;
-use crate::utils::path_utils::{find_dart_package_dir, path_to_string};
+use crate::utils::path_utils::find_dart_package_dir;
+
 use anyhow::Result;
 use include_dir::{include_dir, Dir};
 use itertools::Itertools;
-use log::{debug, warn};
+use log::{debug, info, warn};
 use std::env;
 use std::ffi::OsStr;
 use std::path::{Path, PathBuf};
 
-static INTEGRATION_TEMPLATE_DIR: Dir<'_> =
-    include_dir!("$CARGO_MANIFEST_DIR/assets/integration_template");
+use super::utils::ReplaceContentConfig;
+
+static SHARED_INTEGRATION_TEMPLATE_DIR: Dir<'_> =
+    include_dir!("$CARGO_MANIFEST_DIR/assets/integration_template/shared");
+
+static APP_INTEGRATION_TEMPLATE_DIR: Dir<'_> =
+    include_dir!("$CARGO_MANIFEST_DIR/assets/integration_template/app");
+
+static PLUGIN_INTEGRATION_TEMPLATE_DIR: Dir<'_> =
+    include_dir!("$CARGO_MANIFEST_DIR/assets/integration_template/plugin");
 
 pub struct IntegrateConfig {
     pub enable_integration_test: bool,
     pub enable_local_dependency: bool,
     pub rust_crate_name: Option<String>,
     pub rust_crate_dir: String,
+    pub template: ProjectType,
```

Comment:
nit: since we rename to template, maybe rename

* ProjectType
* ProjectTypeArg

to template as well